src/components/shared/misc/SchedulerEditor.vue

```diff
@@ -0,0 +1,506 @@
+
+                const index = this.alternating_start % 2 ? 1 : 0
+                const altIndex = this.alternating_start % 2 ? 0 : 1
+
+                for (let i = index; i < 12; i += 2) {
+                    month = moment().month(i)
+                    months[month.format('MMMM').toLowerCase()] = true
+                }
+
+                for (let i = altIndex; i < 12; i += 2) {
+                    month = moment().month(i)
+                    months[month.format('MMMM').toLowerCase()] = false
+                }
+
+                this.schedule.frequency = defaultsDeep({ months }, this.schedule.frequency)
+            },
+
+            setAllWorking() {
+                this.schedule.frequency = defaultsDeep({
+                    days: {
+                        monday: true,
+                        tuesday: true,
+                        wednesday: true,
+                        thursday: true,
+                        friday: true,
+                        saturday: false,
+                        sunday: false,
+                    },
+                }, this.schedule.frequency)
+            },
+
+            setAll() {
+                this.setAllWorking()
+                this.$nextTick(() => {
+                    this.schedule.frequency = defaultsDeep({
+                        days: {
+                            monday: true,
+                            tuesday: true,
+                            wednesday: true,
+                            thursday: true,
+                            friday: true,
+                            saturday: true,
+                            sunday: true,
+                        },
+                    }, this.schedule.frequency)
+                })
+            },
+
+            clearAll() {
+                this.setAllWorking()
+                this.$nextTick(() => {
+                    this.schedule.frequency = defaultsDeep({
+                        days: {
+                            monday: false,
+                            tuesday: false,
+                            wednesday: false,
+                            thursday: false,
+                            friday: false,
+                            saturday: false,
+                            sunday: false,
+                        },
+                    }, this.schedule.frequency)
+                })
+            },
+
+            clearEndDate() {
+                this.$refs.endDate.picker.setDate(null)
+                this.schedule.limit.endsAt = ''
+            },
+
+            done() {
+                this.$emit('schedule-set', this.getSchedulerObject)
+                this.$nextTick(this.getSchedule())
+            },
+        },
+
+        mounted() {
+            this.getSchedule()
+            this.$watch('getStartDate', this.$refs.endDate.create)
+            this.$refs.startDate.$refs.pickerfield.setAttribute('readonly', 'true')
+            this.$refs.endDate.$refs.pickerfield.setAttribute('readonly', 'true')
```

Comment:
Should I update this in the croud-date-picker component so it updates for everything using that component?

Reply:
A read only prop for the field?, i wasn't sure where else we would use it but we could add it if you prefer.

Reply:
What problem does this code fix?